csrank/objectranking/lambdamart.py

```diff
@@ -0,0 +1,376 @@
+            result[-1].append(s)
+        result = list(map(np.array, result))
+        return result
+    
+    def fit(self, X, y, **kwargs):
+        """
+            Fit a LambdaMART algorithm to the provided X and y arrays where X contains the features and y being the relevance scores.
+
+            Parameters
+            ----------
+            X : numpy array
+                (n_instances, n_objects, n_features)
+                Feature vectors of the objects
+            Y : numpy array
+                (n_instances, n_objects)
+                Rankings of the given objects
+            **kwargs
+                Keyword arguments for the fit function
+            
+            Returns
+            -------
+            Returns the model which is in turn just a list of all the trees that make up the MART model
+
+        """
+        #check the case if the ensemble already has some trees then clear the trees so that the trees from the previous iteration are not used.
+        if len(self.ensemble) > 0:
+            self.ensemble.clear()
+        
+        train_file = self._prepare_train_data(X, y)
+        scores = train_file[:, 0]
+        queries = train_file[:, 1]
+        features = train_file[:, 3:]       
+
+        model_preds = np.zeros(len(features))
+        
+        for i in range(self.number_of_trees):
+            #print(" Iteration: " + str(i + 1))
+            true_data = self._group_by_queries(scores, queries)
+            model_data = self._group_by_queries(model_preds, queries)
+
+            with Pool(self.num_process) as pool:
+                lambdas_draft = pool.map(query_lambdas, list(zip(true_data, model_data)))
+                lambdas = list(chain(*lambdas_draft))
+
+            tree = DecisionTreeRegressor(criterion=self.criterion,
+                                         splitter=self.splitter,
+                                         max_depth=self.max_depth,
+                                         min_samples_split=self.min_samples_split,
+                                         min_samples_leaf=self.min_samples_leaf,
+                                         min_weight_fraction_leaf=self.min_weight_fraction_leaf,
+                                         max_features=None,
+                                         random_state=self.random_state,
+                                         max_leaf_nodes=self.max_leaf_nodes,
+                                         min_impurity_decrease=self.min_impurity_decrease,
+                                         min_impurity_split=self.min_impurity_split)
+            tree.fit(features, lambdas)
+
+            self.ensemble.append(tree)
+
+            prediction = tree.predict(features)
+            model_preds += self.learning_rate * prediction
+            #TODO: Remove the next two statements after debugging
+            train_score = self._score(model_preds, scores, queries, 10)
+            print("  --iteration train score " + str(train_score))
```

Comment:
Do not use print statements in production code.

Either properly setup your code with [logging](https://docs.python.org/3/howto/logging.html) or simply remove it.